devicetypes/smartthings/tint-remote-controller.src/tint-remote-controller.groovy

```diff
@@ -0,0 +1,308 @@
+	state.onOffValue = (state.onOffValue == "off" ? "on" : "off")
+	createEvent(name: "switch", value: state.onOffValue)
+}
+
+private getColorEvent(descMap) {
+	def CurrentX = Integer.parseInt(descMap.data[1] + descMap.data[0], 16)
+	def CurrentY = Integer.parseInt(descMap.data[3] + descMap.data[2], 16)
+	def HS = color_xyYtoHSV(CurrentX, CurrentY)
+	return [
+		createEvent(name: "hue", value: HS.H * 100),
+		createEvent(name: "saturation", value: 100)
+	]
+}
+
+private getColorTemperatureEvent(descMap) {
+	def miredTemp = Integer.parseInt(descMap.data[1] + descMap.data[0], 16)
+	def colorTemperature = 1_000_000 / miredTemp
+	colorTemperature = colorTemperature < 2750 ? 2700 : (colorTemperature > 6500 ? 6500 : colorTemperature)
+	return createEvent(name: "colorTemperature", value: colorTemperature)
+}
+
+private getLevelEvent(descMap) {
+	def diff = descMap.data[0] == "00" ? 10 : -10 //0x00 Level up / 0x01 Level down
+	state.currentLevel = state.currentLevel + diff
+	state.currentLevel = Math.min(state.currentLevel, 100)
+	state.currentLevel = Math.max(state.currentLevel, 1)
+	return createEvent(name: "level", value: state.currentLevel)
+}
+
+private getButtonEvent(descMap) {
+	def childId = getButtonNumber(descMap.data[3])
+	def description = "Button no. ${childId} was ${value}"
+	def event = createEvent(name: "button", value: "pushed", descriptionText: description, data: [buttonNumber: childId], isStateChange: true)
+	sendEventToChild(childId, event)
+	return event
+}
+
+def on() {
+	state.onOffValue = "on"
+	createEvent(name: "switch", value: "on")
+}
+
+def off() {
+	state.onOffValue = "off"
+	createEvent(name: "switch", value: "off")
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	return zigbee.readAttribute(zigbee.ONOFF_CLUSTER, ON_OFF_ATTRIBUTE)
+}
+
+def configure() {
+	def groupsList = [0x4004, 0x4005, 0x4006]
+		groupsList.each {
+			sendHubCommand(addHubToGroup(it))
+		}
+	sendEvent(name: "checkInterval", value: (2 * 60 + 10) * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	return zigbee.onOffConfig(5, 3600)
+}
+
+def setColorTemperature(value) {
+	sendEvent(name: "colorTemperature", value: value)
+}
+
+def setLevel(value) {
+	state.currentLevel = value
+	sendEvent(name: "level", value: state.currentLevel)
+}
+
+def setColor(value){
+	sendEvent(name: "hue", value: value.hue)
+	sendEvent(name: "saturation", value: value.saturation)
+}
+
+def installed() {
+	def numberOfButtons = 6
+	sendEvent(name: "numberOfButtons", value: numberOfButtons, displayed: false)
+	addChildButtons(numberOfButtons)
+	state.currentLevel = 100
+	sendEvent(name: "level", value: 100, displayed: false)
+	state.onOffValue = "off"
+	createEvent(name: "switch", value: "off", displayed: false)
+	createEvent(name: "colorTemperature", value: 2700, displayed: false)
+	sendEvent(name: "saturation", value: 100, descriptionText: "Color has changed", displayed: false)
+}
+
+private color_xyYtoHSV(CurrentX, CurrentY) { 
```

Comment:
just checking: xxY is a different color space than the one @dkirker did for ikea in zll-rgb-bulb, right?

Reply:
xyY is xyz but in only 2D (http://www.colorwiki.com/wiki/XyY). I believe the function that I wrote in zll-rgb-bulb is xyY.

Keep in mind, the table M in the XyToRgb and RgbToXy functions may actually differ per device (so this might be what you would need to tweak so that you can use the saturation value instead of hard-coding to 100).

Another thing to keep in mind from a comment that I put in zll-rgb-bulb: "Not all XY color values are representable in rgb", and the conversion is not lossless.

@PKacprowiczS We should keep the two device handlers consistent with what they use. There is also an implementation of these conversions in HubCore. If the implementation has a bug, then it should be fixed in this PR, noted in the comments, and then a PR created to fix it in the other DTH as well as a ticket created so that we can fix it in HubCore.



Let's use the logic from zll-rgb-bulb for consistency. Though, we should use colorUtil.hslToRgb(h, s, l) and colorUtil.rgbToHsl(r, g, b) (we can probably even update zll-rgb-bulb and rgbw-light to use those, and that might help with a couple of bugs reported in zll-rgb-bulb's color conversion).

Reply:
Based on my work with the device, I wouldn't go into much trouble with tweaking M matrix, as sending hardcoded 100 saturation might not look pretty, I'd say it's still doing it's job in creating accurate events. Although, I'll elaborate on this matter more on JIRA, cause I believe it would be easier to express myself with a help of a video.